RecoHGCal/TICL/plugins/LinkingAlgoByDirectionGeometric.cc

```diff
@@ -0,0 +1,611 @@
+                    SimpleDiskBounds(rMinMax_interface.first, rMinMax_interface.second, zSide - 0.5, zSide + 0.5))
+            .get());
+  }
+}
+
+void LinkingAlgoByDirectionGeometric::linkTracksters(const edm::Handle<std::vector<reco::Track>> tkH,
+                                                     const edm::ValueMap<float> &tkTime,
+                                                     const edm::ValueMap<float> &tkTimeErr,
+                                                     const edm::ValueMap<float> &tkTimeQual,
+                                                     const std::vector<reco::Muon> &muons,
+                                                     const edm::Handle<std::vector<Trackster>> tsH,
+                                                     std::vector<TICLCandidate> &resultLinked) {
+  constexpr double mpion = 0.13957;
+  constexpr float mpion2 = mpion * mpion;
+
+  const auto &tracks = *tkH;
+  const auto &tracksters = *tsH;
+
+  auto bFieldProd = bfield_.product();
+  const Propagator &prop = (*propagator_);
+
+  // propagated point collections
+  // elements in the propagated points collecions are used
+  // to look for potential linkages in the appropriate tiles
+  std::vector<std::pair<Vector, unsigned>> trackPColl;     // propagated track points and index of track in collection
+  std::vector<std::pair<Vector, unsigned>> tkPropIntColl;  // tracks propagated to lastLayerEE
+  std::vector<std::pair<Vector, unsigned>> tsPropIntColl;  // Tracksters in CE-E, propagated to lastLayerEE
+  std::vector<std::pair<Vector, unsigned>> tsHadPropIntColl;  // Tracksters in CE-H, propagated to lastLayerEE
+
+  // tiles, element 0 is bw, 1 is fw
+  std::array<TICLLayerTile, 2> tracksterPropTiles = {};  // all Tracksters, propagated to layer 1
+  std::array<TICLLayerTile, 2> tsPropIntTiles = {};      // all Tracksters, propagated to lastLayerEE
+  std::array<TICLLayerTile, 2> tsHadPropIntTiles = {};   // Tracksters in CE-H, propagated to lastLayerEE
+
+  // filters (true for) anything but EM
+  auto isHadron = [&](const Trackster &t) -> bool {
+    auto cumulative_prob = 0.;
+    for (const auto index : filter_on_categories_) {
+      cumulative_prob += t.id_probabilities(index);
+    }
+    return ((cumulative_prob <= pid_threshold_) and (t.raw_em_energy() != t.raw_energy())) or
+           (t.raw_em_energy() < energy_em_over_total_threshold_ * t.raw_energy());
+  };
+  if (LinkingAlgoBase::algo_verbosity_ > LinkingAlgoBase::Advanced)
+    LogDebug("LinkingAlgoByDirectionGeometric") << "------- Geometric Linking ------- \n";
+
+  // Propagate tracks
+  std::vector<unsigned> candidateTrackIds;
+  for (unsigned i = 0; i < tracks.size(); ++i) {
+    const auto &tk = tracks[i];
+
+    reco::TrackRef trackref = reco::TrackRef(tkH, i);
+    // also veto tracks associated to muons
+    int muId = PFMuonAlgo::muAssocToTrack(trackref, muons);
+    if (LinkingAlgoBase::algo_verbosity_ > LinkingAlgoBase::Advanced)
```

Comment:
it seems strange to protect a compile-time-enabled message with a runtime flag.

Reply:
Hi! We use this strategy to have different log-debug levels in TICL, but if this is a problem I can remove it